src/Rules.hs

```diff
@@ -98,12 +98,7 @@ packageRules = do
     let contexts        = liftM3 Context        allStages knownPackages allWays
         vanillaContexts = liftM2 vanillaContext allStages knownPackages
 
-    forM_ contexts $ mconcat
-        [ Rules.Compile.compilePackage readPackageDb
-        , Rules.Library.buildPackageLibrary ]
-
-    let dynamicContexts = liftM3 Context [Stage1 ..] knownPackages [dynamic]
-    forM_ dynamicContexts Rules.Library.buildDynamicLib
+    forM_ contexts (Rules.Compile.compilePackage readPackageDb)
```

Comment:
Can we add a TODO here with a link to this PR to eventually convert other build rules into the new approch? I think it's more direct and robust and we should be aiming to refactor all rules analogously.

Reply:
Sure thing, will do in an upcoming patch.